crates/hir_def/src/item_tree.rs

```diff
@@ -304,6 +306,32 @@ impl TypeRefStorage {
     }
 }
 
+/// `TraitRef` interner.
+#[derive(Default, Debug, Eq, PartialEq)]
+struct TraitRefStorage {
+    arena: Arena<Arc<TraitRef>>,
+    map: FxHashMap<Arc<TraitRef>, Idx<Arc<TraitRef>>>,
+}
```

Comment:
After this PR, we need to check if we can add `InterningArena` to `la-arena` or just to this module.